src/MSBuild.UnitTests/CommandLineSwitches_Tests.cs

```diff
@@ -295,6 +295,29 @@ public void PropertySwitchIdentificationTests(string property)
             unquoteParameters.ShouldBeTrue();
         }
 
+        [Theory]
+        [InlineData("restoreproperty")]
+        [InlineData("RESTOREPROPERTY")]
+        [InlineData("RestoreProperty")]
+        [InlineData("rp")]
+        [InlineData("RP")]
+        public void RestorePropertySwitchIdentificationTests(string property)
+        {
+            CommandLineSwitches.ParameterizedSwitch parameterizedSwitch;
+            string duplicateSwitchErrorMessage;
+            bool multipleParametersAllowed;
+            string missingParametersErrorMessage;
+            bool unquoteParameters;
+            bool emptyParametersAllowed;
+
+            CommandLineSwitches.IsParameterizedSwitch(property, out parameterizedSwitch, out duplicateSwitchErrorMessage, out multipleParametersAllowed, out missingParametersErrorMessage, out unquoteParameters, out emptyParametersAllowed).ShouldBeTrue();
+            parameterizedSwitch.ShouldBe(CommandLineSwitches.ParameterizedSwitch.RestoreProperty);
+            duplicateSwitchErrorMessage.ShouldBeNull();
+            multipleParametersAllowed.ShouldBeTrue();
+            missingParametersErrorMessage.ShouldNotBeNull();
```

Comment:
Maybe specify that this should be the property error message? You would have to get it out of strings with ResourceUtilities.FormatResourceString or something to ensure it works on any locale.

Reply:
There is a set of these unit tests which are all effectively testing the `CommandLineSwitches.IsParameterizedSwitch()` method. That method doesn't retrieve the resource string. It pulls a `ParameterizedSwitchInfo` object from the `s_parameterizedSwitchesMap` array.

The `RestorePropertySwitchIdentificationTests` test could be modified to check that the value of `missingParametersErrorMessage` is `"MissingPropertyError"`.

Should all of the unit tests with `missingParametersErrorMessage.ShouldNotBeNull();` be updated accordingly? (I'm happy to make the changes but I don't want to presume.)

Reply:
That sounds good to me. As far as changing any of the other tests, it sounds reasonable to me, but it's less relevant to the main point of this PR, so if you decide to change them all, I'd put that in a separate PR.

Reply:
The unit test has been updated. Thanks